### DISCUSSION TITLE
vbank layering doesn't make sense to me

### DISCUSSION THREAD

**Discussion by SRB2er** [nesbox/TIC-80#2550] [Q&A] [answered]
So from what I understand, vbank allows you to swap to a separate palette (and back) on the same frame, allowing for 31 colors (or 32 if you don't want transpareny) per scanline.
..but then it also says this:

> The banks are rendered to the screen simultaneously with any opaque portions of bank 1 appearing on "top" of bank 0. Bank 0 data is rendered using the bank 0 palette and bank 1 is rendered using the bank 1 palette. OVR TRANSPARENCY (0x03FF8) controls which palette index of bank 1 is considered transparent. Currently this behavior cannot be modified.

This part...confuses me.
If you want to draw on top of something, draw it _after._
I don't understand what makes this part special about vbank(1)
Can someone explain it better?

**Comment by Weeppiko** [comment, marked as answer]
`vbank` doesn't change a palette, it swaps the layer that does have it's own palette: `vbank(0)` is the lower layer and `vbank(1)` is the upper layer. Everything drawn on `vbank(1)` will be on top of `vbank(0)`, and since it is on top, one of the colors need to be treated as transparency.

**Comment by Weeppiko** [comment]
One example of `vbank` usage would be using scanline effects only in one of the layers, like this (background on `vbank(0)` and sprites on `vbank(1)`):
![video2](https://github.com/nesbox/TIC-80/assets/88540876/2a5bfde9-f733-42c7-b0de-607100c58d50)
